backend/module/chaos/serverexperimentation/xds/xds_integration_test.go

```diff
@@ -111,3 +122,40 @@ func awaitExpectedReturnValueForSimpleCall(t *testing.T, e *envoytest.EnvoyHandl
 
 	return nil
 }
+
+func TestEnvclearoyECDSFaults(t *testing.T) {
+	xdsConfig := &xdsconfigv1.Config{
+		RtdsLayerName:             "rtds",
+		CacheRefreshInterval:      ptypes.DurationProto(time.Second),
+		IngressFaultRuntimePrefix: "fault.http",
+		EgressFaultRuntimePrefix:  "egress",
+		EcdsAllowList:             &xdsconfigv1.Config_ECDSAllowList{EnabledClusters: []string{"test-cluster"}},
+	}
+
+	ts := xdstest.NewTestModuleServer(New, true, xdsConfig)
+	defer ts.Stop()
+
+	e, err := envoytest.NewEnvoyHandle()
+	assert.NoError(t, err)
+
+	code, err := e.MakeSimpleCall()
+	assert.NoError(t, err)
+	assert.Equal(t, 503, code)
+
+	experiment := createTestExperiment(t, 404, ts.Storer)
+
+	err = awaitExpectedReturnValueForSimpleCall(t, e, awaitReturnValueParams{
+		timeout:        2 * time.Second,
+		expectedStatus: 404,
+	})
+	assert.NoError(t, err, "did not see faults enabled")
+
+	// Not testing TTL by stopping the server as its currently not supported for ECDS in Upstream Envoy
```

Comment:
This comment seems a bit odd because it assumes the reader was expecting to see a TTL test here which they probably don't, maybe reword a bit? Maybe just turn this into a TODO